### PR TITLE
PR: Add checklists for Github issue report and pull requests

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Followed [PEP8](https://www.python.org/dev/peps/pep-0008/) for code style
* [x] Ensured your pull request hasn't eliminated unrelated blank lines/spaces,
      modified the ``spyder/defaults`` directory, or added new icons/assets
* [x] ~~Wrote at least one-line docstrings for any new functions~~
* [x] ~~Added at least one unit test covering the changes, if at all possible~~
* [x] Described your changes and the motivation for them below
* [x] ~~Noted what issue(s) this pull request resolves, creating one if needed~~
* [x] Included a screenshot, if this PR makes any visible changes to the UI

## Description of Changes

This PR adds a checklist for the user to complete before submitting an issue report (both in the issue template and submitted from the Spyder Help menu; automatic/semi-auto error reports are unaffected), to hopefully help increase the quality of issues received and enable speedy closure of those that aren't helpful. To avoid any chance of an incomplete report due to length limitations, the text verbosity for the Help menu report (particularly the mostly redundant reminder) was cut down where possible.

image

Also, it adds a heavily revised PR template (as used here) with a checklist (top), along with formatting enhancements, some additional items and helpful tips in the comments.

### Issue(s) Resolved

As this is an internal change to the Github templates, none believed needed.

### CODE REVIEW

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Need anything from me to get this merged?

**Comment by CAM-Gerlach** [comment]
@ccordoba12 ?

**Comment by ccordoba12** [comment]
I think there are way too many boxes to check. Could you reduce the list a bit?

**Comment by CAM-Gerlach** [comment]
> I think there are way too many boxes to check. Could you reduce the list a bit?

Okay, sure. I reduced the number of boxes in the issue report template by 1/3, and considerably cut down most of the text. The Help menu semi automated report version is even shorter still. Also, I cut the pull request one down by 1/4 and similarly eliminated some extra text. Let me know what you think,

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -25,13 +37,16 @@ PASTE TRACEBACK HERE
...
-* Spyder:
-* Python:
-* Qt:
-* PyQt:
-* Operating System:
+* Spyder version:
+* Python version:
+* Qt version:
+* PyQt version:
+* OS name and version:
```
Comment (ccordoba12) [inline .github/ISSUE_TEMPLATE.md]: `OS` is a term some people could not understand. So please revert this to `Operating System:`

File: spyder/app/mainwindow.py [hunk 2]
```diff
@@ -2420,34 +2420,53 @@ def render_issue(self, description='', traceback=''):
...
 
-## Package Versions
+## Versions
 
-* Spyder version: {spyder_version} {commit}
-* Python version: {python_version}
-* Qt version: {qt_version}
-* {qt_api_name} version: {qt_api_version}
-* Operating system: {os_name} {os_version}
+* Spyder ver: {spyder_version} {commit}
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: I think we have talked about this before. `ver` can be confusing and is not explicit enough for non-English speakers. So please revert this to `version`.

File: spyder/app/mainwindow.py [hunk 3]
```diff
@@ -2420,34 +2420,53 @@ def render_issue(self, description='', traceback=''):
...
+* Python ver: {python_version}
+* Qt ver: {qt_version}
+* {qt_api_name} ver: {qt_api_version}
+* OS: {os_name} {os_version}
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: Please revert this to `Operating System`.

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (CAM-Gerlach) [reply .github/ISSUE_TEMPLATE.md]: The problem (and I think I mentioned this before as well) is that we're out of characters to play around with, as aside from a small buffer I built in for one or two lines more of deps, the report starts getting cut off or outright producing errors when trying to send it to github, and I already cut and cut and cut the issue checklist down to under 1/3 of its original length as well as the disclaimer, headers etc . The real issue is that the dependency list takes up ~3-4x the characters it should due to all the fancy symbols used needed percent encoding; even a small change here (eliminating one or two deps that rarely matter; using one or two unnecessary symbols in the plain text version, etc. will cut dozens or even hundreds of characters from the final URL.

Please advise what you'd like me to do—fix the plain text display of the deps? Convert this to use the new error dialog? Just risk it and add back the chars? If I do the latter, adding even one more dep (as is discussed elsewhere) will likely break the whole URL based error reporting mechanism, or at least come dangerously close.

File: spyder/app/mainwindow.py (hunk 2 above)
Comment (CAM-Gerlach) [reply spyder/app/mainwindow.py]: I agree, explicit is better than implicit, and I don't often use abbreviations myself. See above for the reasons and alternatives.

File: spyder/app/mainwindow.py (hunk 3 above)
Comment (CAM-Gerlach) [reply spyder/app/mainwindow.py]: See abpve/

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (CAM-Gerlach) [reply .github/ISSUE_TEMPLATE.md]: Let me know what you'd like me to do about this, @ccordoba12 . Thanks!

Comment (ccordoba12) [reply .github/ISSUE_TEMPLATE.md]: Sorry for the delay in coming back to you. Please remove this entry in the new list:

> Completed the problem description, steps to reproduce, ...

That's kind of obvious and it should be enough to add the characters that we need.

Comment (CAM-Gerlach) [reply .github/ISSUE_TEMPLATE.md]: Sorry for the confusion; I meant to make the above initial comment on the mainwindow review item instead. That entry only exists (in that lengthy form at least) in the issue report template; the relevant text for the Help menu Report Issue item is in ``mainwindow``. However, 

> Completed all applicable sections below

while much shorter, does exist and, once percent encoded, is indeed around the same length as a dependency line, much longer than "versions". Unfortunately it isn't as obvious as one would think as many reporters don't do it, but at least it is something we can easily see for ourselves. I'll go ahead and eliminate it and spell out versions and OS, and push the changes momentarily; that should at least give us a little breathing room for the time being.

**Pull request merged**